docs/DeveloperGuide.md

```diff
@@ -87,6 +123,34 @@ This section contains links to other relevant guides.
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Add task feature
+The adding of task mechanism is facilitated by `LogicManager`.
+
+The retrieving of task details is done by `Parser`. It splits the user input based on command type and the various parameters' description regardless of the order the user input the parameters in.
```

Comment:
> It splits the user input based on command type...

`based on the command type` instead of `based on command type`?

Just a suggestion: I think it is easier to read if we say "the user can provide the parameters in any order" in a separate sentence.